typings/index.d.ts

```diff
@@ -26,6 +26,17 @@ export interface DownshiftProps {
   inputValue?: string
   highlightedIndex?: number
   id?: string
+  environment?: Environment
+}
+
+export interface Environment {
+  addEventListener: (type: string, listener: (event: Event) => void) => void
+  removeEventListener: (type: string, listener: () => void) => void
```

Comment:
For both `addEventListener` and `removeEventListener`, I suggest just using `Function`  as the callback type (as the rest of this file does). This gives us the most flexibility.